sqlglot/dialects/dialect.py

```diff
@@ -412,6 +412,16 @@ def min_or_least(self: Generator, expression: exp.Min) -> str:
     return rename_func(name)(self, expression)
 
 
+def count_if_to_sum(self: Generator, expression: exp.CountIf) -> str:
+    cond = expression.this
+
+    if isinstance(expression.this, exp.Distinct):
+        cond = expression.this.expressions[0]
+        self.unsupported("DISTINCT is not supported when converting COUNT_IF to SUM")
+
+    return self.sql(exp.func("sum", exp.func("if", cond, 1, 0)))
```

Comment:
return self.func("sum", self.func("if", cond, 1, 0))

Reply:
I don't think we can do this here because we need a string, right?

Reply:
doesn’t self.func return a string?

Reply:
Nope, returns an `exp.Func`

Reply:
https://github.com/tobymao/sqlglot/blob/main/sqlglot/generator.py#L1859

are you sure?

Reply:
ignore me, making the fix

Reply:
I didn't change the internal `exp.func` because `if` needs to be transpiled correctly across dialects (e.g. sqlite supports `IIF` instead of `IF`).